CP-30787: Crypto code build&linking changes

From: Edwin Török <edvin.torok@citrix.com>

if secureboot is not enabled disable all Crypto.
if it is enabled then disable the following:
disable DSA/ECC X509 signature algorithms
disable MD4, Ciphers, RsaExt, Pkcs7Sign, and Dh

Signed-off-by: Edwin Török <edvin.torok@citrix.com>


[anthony] rebased:
MD4, ARC4, TDES removed upstreamed.
---
 CryptoPkg/Library/BaseCryptLib/BaseCryptLib.inf | 10 +++++-----
 CryptoPkg/Library/BaseCryptLib/SmmCryptLib.inf  |  2 +-
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/CryptoPkg/Library/BaseCryptLib/BaseCryptLib.inf b/CryptoPkg/Library/BaseCryptLib/BaseCryptLib.inf
index 3d7b917103..78da5f0d6b 100644
--- a/CryptoPkg/Library/BaseCryptLib/BaseCryptLib.inf
+++ b/CryptoPkg/Library/BaseCryptLib/BaseCryptLib.inf
@@ -37,16 +37,16 @@
   Hash/CryptParallelHashNull.c
   Hmac/CryptHmacSha256.c
   Kdf/CryptHkdf.c
-  Cipher/CryptAes.c
+  Cipher/CryptAesNull.c
   Pk/CryptRsaBasic.c
-  Pk/CryptRsaExt.c
+  Pk/CryptRsaExtNull.c
   Pk/CryptPkcs1Oaep.c
-  Pk/CryptPkcs5Pbkdf2.c
-  Pk/CryptPkcs7Sign.c
+  Pk/CryptPkcs5Pbkdf2Null.c
+  Pk/CryptPkcs7SignNull.c
   Pk/CryptPkcs7VerifyCommon.c
   Pk/CryptPkcs7VerifyBase.c
   Pk/CryptPkcs7VerifyEku.c
-  Pk/CryptDh.c
+  Pk/CryptDhNull.c
   Pk/CryptX509.c
   Pk/CryptAuthenticode.c
   Pk/CryptTs.c
diff --git a/CryptoPkg/Library/BaseCryptLib/SmmCryptLib.inf b/CryptoPkg/Library/BaseCryptLib/SmmCryptLib.inf
index 91a1715095..18327243da 100644
--- a/CryptoPkg/Library/BaseCryptLib/SmmCryptLib.inf
+++ b/CryptoPkg/Library/BaseCryptLib/SmmCryptLib.inf
@@ -44,7 +44,7 @@
   Hash/CryptParallelHash.c
   Hmac/CryptHmacSha256.c
   Kdf/CryptHkdfNull.c
-  Cipher/CryptAes.c
+  Cipher/CryptAesNull.c
   Pk/CryptRsaBasic.c
   Pk/CryptRsaExtNull.c
   Pk/CryptPkcs1Oaep.c
-- 
2.43.0

